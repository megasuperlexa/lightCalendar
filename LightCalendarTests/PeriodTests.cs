using ApprovalTests;
using ApprovalTests.Reporters;
using LightCalendar;
using System;
using System.Globalization;
using System.Linq;
using System.Threading;
using Xunit;

[assembly:UseReporter(typeof(DiffReporter))]

namespace LightCalendarTests
{
    public class PeriodTests
    {
        public PeriodTests()
        {
            CultureInfo newCulture = (CultureInfo)Thread.CurrentThread.CurrentCulture.Clone();
            newCulture.DateTimeFormat.ShortDatePattern = "MM/dd/yyyy";
            Thread.CurrentThread.CurrentCulture = newCulture;
        }
        
        [Fact]
        public void DayCountTest()
        {
            // Arrange
            var now = DateTime.UtcNow;
            var today = new Period(now, now);
            var todayTomorrow = new Period(now, now.AddDays(1));
            var todayYesterday = new Period(now, now.AddDays(-1));
            // Act
            // Assert
            Assert.False(today.IsEmpty);
            Assert.Equal(1, today.GetDayCount());
            
            Assert.Equal(2, todayTomorrow.GetDayCount()); // count inclusive
            
            Assert.True(todayYesterday.IsEmpty); // Empty period that ended before beginning
            Assert.Equal(0, todayYesterday.GetDayCount());
            
            Assert.Empty(todayYesterday.GetDailySchedule());
        }

        [Fact]
        public void MonthCountTest()
        {
            // Arrange
            var p = new Period(new DateTime(2015, 10, 7), new DateTime(2016, 10, 7));
            var p2 = new Period(new DateTime(2015, 10, 1), new DateTime(2015, 11, 1));
            var p3 = new Period(new DateTime(2015, 10, 1), new DateTime(2015, 10, 31));
            // Assert
            Assert.Equal(12, p.GetMonthCount());
            Assert.Equal(1, p2.GetMonthCount());
            Assert.Equal(1, p3.GetMonthCount());
        }

        [Fact]
        public void EqualityTest()
        {
            var p1 = new Period(new DateTime(2017, 10, 2), new DateTime(2017, 10, 20));
            var p2 = new Period(new DateTime(2017, 10, 2), new DateTime(2017, 10, 20));
            var p3 = (new DateTime(2017, 10, 2), new DateTime(2017, 10, 20));
            var p4 = (new DateTime(2016, 10, 2), new DateTime(2017, 10, 20));

            Assert.Equal(p1, p2);
            Assert.Equal(p1, p3);
            Assert.NotEqual(p1, p4);

            var f1 = FiscalYear.GetFiscalYear(new DateTime(2017, 10, 2), new DateTime(2017, 10, 3));
            // for fiscal year, only month and day are relevant for fiscal year end
            var f2 = FiscalYear.GetFiscalYear(new DateTime(2010, 10, 2), new DateTime(2017, 10, 3));

            Assert.Equal(f1, f2);
        }
        
        [Fact]
        public void OverlapTest()
        {
            // Arrange
            var now = DateTime.UtcNow;
            var today = new Period(now, now);
            var todayTomorrow = new Period(now, now.AddDays(1));
            var todayYesterday = new Period(now, now.AddDays(-1));
            var yesterdayTomorrow = new Period(now.AddDays(-1), now.AddDays(1));
            var monthAheadPlusDay = new Period(now.AddMonths(1), now.AddMonths(1).AddDays(1));
            var monthBackwardsAndDay = new Period(now.AddMonths(-1), now.AddMonths(-1).AddDays(1));
            // Act
            var overlap1 = today.Overlap(todayTomorrow);
            var overlapWithEmpty = todayTomorrow.Overlap(todayYesterday);
            var overlap2 = todayTomorrow.Overlap(monthAheadPlusDay);
            var overlap3 = todayTomorrow.Overlap(monthBackwardsAndDay);
            var overlap4 = yesterdayTomorrow.Overlap(todayTomorrow);
            // Assert
            Assert.False(overlap1.IsEmpty);
            Assert.Equal(1, overlap1.GetDayCount());
            Assert.True(overlapWithEmpty.IsEmpty); // overlaping with empty
            Assert.True(overlap2.IsEmpty); // overlaping separate periods
            Assert.True(overlap3.IsEmpty);
            Assert.Equal(2, overlap4.GetDayCount());
            Assert.True(yesterdayTomorrow.Contains(todayTomorrow));
        }
        
        [Fact]
        public void CombineTest()
        {
            // Arrange
            var now = DateTime.UtcNow;
            var todayTomorrow = new Period(now, now.AddDays(1));
            var fourDays = new Period(now.AddDays(1), now.AddDays(4));
            // Act
            var combine1 = todayTomorrow.Combine(fourDays);
            // Assert
            Assert.Equal(todayTomorrow.Begin, combine1.Begin);
            Assert.Equal(fourDays.End, combine1.End);
            Assert.Equal(5, combine1.GetDayCount());
        }
        
        [Fact]
        public void ExcludeTest()
        {
            // Arrange
            var now = new DateTime(2000, 06, 15);
            var todayTomorrow = new Period(now, now.AddDays(1));
            var fourDays = new Period(now, now.AddDays(4));
            // Act
            var excluded = fourDays.Exclude(todayTomorrow);
            // Assert
            Approvals.VerifyAll(excluded.GetDailySchedule(), "Day");
            
        }
        
        [Fact]
        public void FiscalYearTest()
        {
            // Arrange
            var fy = FiscalYear.GetFiscalYear(new DateTime(2000,06,15), new DateTime(2020,11,1));
            
            Approvals.VerifyAll(fy.GetDailySchedule(), "Day");
        }

        [Fact]
        public void PeriodRatesTest()
        {
            var fy = FiscalYear.GetFiscalYear(new DateTime(2000,01,01), new DateTime(2020,11,1));
            var periods = fy.GetPeriodValues(new[]
            {
                (new DateTime(2020, 01, 01), .01),
                (new DateTime(2020, 05, 01), .015),
                (new DateTime(2020, 08, 15), .007),
            });
            // Assert
            Approvals.VerifyAll(periods.ToArray(), "Rate");
        }
        
        [Fact]
        public void FederalFyTest()
        {
            //FY 2020 is the budget for October 1, 2019 through September 30, 2020. 
            var fy = FiscalYear.GetFiscalYear(new DateTime(2000,09,30), new DateTime(2020,1,1));
            // Assert
            Assert.Equal(new DateTime(2019, 10, 01),fy.Begin);
            Assert.Equal(new DateTime(2020, 09, 30),fy.End);
        }
        
        [Fact]
        public void SeasonsTest()
        {
            var fy = FiscalYear.GetFiscalYear(new DateTime(2000,01,01), new DateTime(2020,11,1));
            var periods = fy.GetPeriodValues(new[]     
            {
                (new DateTime(2020, 01, 01), "Winter"),
                (new DateTime(2020, 03, 01), "Spring"),
                (new DateTime(2020, 06, 01), "Summer"),
                (new DateTime(2020, 09, 01), "Winter again"),
            });
            
            Approvals.VerifyAll(periods.ToArray(), "Season");
        }
    }
}
