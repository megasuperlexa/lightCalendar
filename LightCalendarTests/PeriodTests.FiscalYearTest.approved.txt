Day[0] = 06/16/2020 12:00:00 AM
Day[1] = 06/17/2020 12:00:00 AM
Day[2] = 06/18/2020 12:00:00 AM
Day[3] = 06/19/2020 12:00:00 AM
Day[4] = 06/20/2020 12:00:00 AM
Day[5] = 06/21/2020 12:00:00 AM
Day[6] = 06/22/2020 12:00:00 AM
Day[7] = 06/23/2020 12:00:00 AM
Day[8] = 06/24/2020 12:00:00 AM
Day[9] = 06/25/2020 12:00:00 AM
Day[10] = 06/26/2020 12:00:00 AM
Day[11] = 06/27/2020 12:00:00 AM
Day[12] = 06/28/2020 12:00:00 AM
Day[13] = 06/29/2020 12:00:00 AM
Day[14] = 06/30/2020 12:00:00 AM
Day[15] = 07/01/2020 12:00:00 AM
Day[16] = 07/02/2020 12:00:00 AM
Day[17] = 07/03/2020 12:00:00 AM
Day[18] = 07/04/2020 12:00:00 AM
Day[19] = 07/05/2020 12:00:00 AM
Day[20] = 07/06/2020 12:00:00 AM
Day[21] = 07/07/2020 12:00:00 AM
Day[22] = 07/08/2020 12:00:00 AM
Day[23] = 07/09/2020 12:00:00 AM
Day[24] = 07/10/2020 12:00:00 AM
Day[25] = 07/11/2020 12:00:00 AM
Day[26] = 07/12/2020 12:00:00 AM
Day[27] = 07/13/2020 12:00:00 AM
Day[28] = 07/14/2020 12:00:00 AM
Day[29] = 07/15/2020 12:00:00 AM
Day[30] = 07/16/2020 12:00:00 AM
Day[31] = 07/17/2020 12:00:00 AM
Day[32] = 07/18/2020 12:00:00 AM
Day[33] = 07/19/2020 12:00:00 AM
Day[34] = 07/20/2020 12:00:00 AM
Day[35] = 07/21/2020 12:00:00 AM
Day[36] = 07/22/2020 12:00:00 AM
Day[37] = 07/23/2020 12:00:00 AM
Day[38] = 07/24/2020 12:00:00 AM
Day[39] = 07/25/2020 12:00:00 AM
Day[40] = 07/26/2020 12:00:00 AM
Day[41] = 07/27/2020 12:00:00 AM
Day[42] = 07/28/2020 12:00:00 AM
Day[43] = 07/29/2020 12:00:00 AM
Day[44] = 07/30/2020 12:00:00 AM
Day[45] = 07/31/2020 12:00:00 AM
Day[46] = 08/01/2020 12:00:00 AM
Day[47] = 08/02/2020 12:00:00 AM
Day[48] = 08/03/2020 12:00:00 AM
Day[49] = 08/04/2020 12:00:00 AM
Day[50] = 08/05/2020 12:00:00 AM
Day[51] = 08/06/2020 12:00:00 AM
Day[52] = 08/07/2020 12:00:00 AM
Day[53] = 08/08/2020 12:00:00 AM
Day[54] = 08/09/2020 12:00:00 AM
Day[55] = 08/10/2020 12:00:00 AM
Day[56] = 08/11/2020 12:00:00 AM
Day[57] = 08/12/2020 12:00:00 AM
Day[58] = 08/13/2020 12:00:00 AM
Day[59] = 08/14/2020 12:00:00 AM
Day[60] = 08/15/2020 12:00:00 AM
Day[61] = 08/16/2020 12:00:00 AM
Day[62] = 08/17/2020 12:00:00 AM
Day[63] = 08/18/2020 12:00:00 AM
Day[64] = 08/19/2020 12:00:00 AM
Day[65] = 08/20/2020 12:00:00 AM
Day[66] = 08/21/2020 12:00:00 AM
Day[67] = 08/22/2020 12:00:00 AM
Day[68] = 08/23/2020 12:00:00 AM
Day[69] = 08/24/2020 12:00:00 AM
Day[70] = 08/25/2020 12:00:00 AM
Day[71] = 08/26/2020 12:00:00 AM
Day[72] = 08/27/2020 12:00:00 AM
Day[73] = 08/28/2020 12:00:00 AM
Day[74] = 08/29/2020 12:00:00 AM
Day[75] = 08/30/2020 12:00:00 AM
Day[76] = 08/31/2020 12:00:00 AM
Day[77] = 09/01/2020 12:00:00 AM
Day[78] = 09/02/2020 12:00:00 AM
Day[79] = 09/03/2020 12:00:00 AM
Day[80] = 09/04/2020 12:00:00 AM
Day[81] = 09/05/2020 12:00:00 AM
Day[82] = 09/06/2020 12:00:00 AM
Day[83] = 09/07/2020 12:00:00 AM
Day[84] = 09/08/2020 12:00:00 AM
Day[85] = 09/09/2020 12:00:00 AM
Day[86] = 09/10/2020 12:00:00 AM
Day[87] = 09/11/2020 12:00:00 AM
Day[88] = 09/12/2020 12:00:00 AM
Day[89] = 09/13/2020 12:00:00 AM
Day[90] = 09/14/2020 12:00:00 AM
Day[91] = 09/15/2020 12:00:00 AM
Day[92] = 09/16/2020 12:00:00 AM
Day[93] = 09/17/2020 12:00:00 AM
Day[94] = 09/18/2020 12:00:00 AM
Day[95] = 09/19/2020 12:00:00 AM
Day[96] = 09/20/2020 12:00:00 AM
Day[97] = 09/21/2020 12:00:00 AM
Day[98] = 09/22/2020 12:00:00 AM
Day[99] = 09/23/2020 12:00:00 AM
Day[100] = 09/24/2020 12:00:00 AM
Day[101] = 09/25/2020 12:00:00 AM
Day[102] = 09/26/2020 12:00:00 AM
Day[103] = 09/27/2020 12:00:00 AM
Day[104] = 09/28/2020 12:00:00 AM
Day[105] = 09/29/2020 12:00:00 AM
Day[106] = 09/30/2020 12:00:00 AM
Day[107] = 10/01/2020 12:00:00 AM
Day[108] = 10/02/2020 12:00:00 AM
Day[109] = 10/03/2020 12:00:00 AM
Day[110] = 10/04/2020 12:00:00 AM
Day[111] = 10/05/2020 12:00:00 AM
Day[112] = 10/06/2020 12:00:00 AM
Day[113] = 10/07/2020 12:00:00 AM
Day[114] = 10/08/2020 12:00:00 AM
Day[115] = 10/09/2020 12:00:00 AM
Day[116] = 10/10/2020 12:00:00 AM
Day[117] = 10/11/2020 12:00:00 AM
Day[118] = 10/12/2020 12:00:00 AM
Day[119] = 10/13/2020 12:00:00 AM
Day[120] = 10/14/2020 12:00:00 AM
Day[121] = 10/15/2020 12:00:00 AM
Day[122] = 10/16/2020 12:00:00 AM
Day[123] = 10/17/2020 12:00:00 AM
Day[124] = 10/18/2020 12:00:00 AM
Day[125] = 10/19/2020 12:00:00 AM
Day[126] = 10/20/2020 12:00:00 AM
Day[127] = 10/21/2020 12:00:00 AM
Day[128] = 10/22/2020 12:00:00 AM
Day[129] = 10/23/2020 12:00:00 AM
Day[130] = 10/24/2020 12:00:00 AM
Day[131] = 10/25/2020 12:00:00 AM
Day[132] = 10/26/2020 12:00:00 AM
Day[133] = 10/27/2020 12:00:00 AM
Day[134] = 10/28/2020 12:00:00 AM
Day[135] = 10/29/2020 12:00:00 AM
Day[136] = 10/30/2020 12:00:00 AM
Day[137] = 10/31/2020 12:00:00 AM
Day[138] = 11/01/2020 12:00:00 AM
Day[139] = 11/02/2020 12:00:00 AM
Day[140] = 11/03/2020 12:00:00 AM
Day[141] = 11/04/2020 12:00:00 AM
Day[142] = 11/05/2020 12:00:00 AM
Day[143] = 11/06/2020 12:00:00 AM
Day[144] = 11/07/2020 12:00:00 AM
Day[145] = 11/08/2020 12:00:00 AM
Day[146] = 11/09/2020 12:00:00 AM
Day[147] = 11/10/2020 12:00:00 AM
Day[148] = 11/11/2020 12:00:00 AM
Day[149] = 11/12/2020 12:00:00 AM
Day[150] = 11/13/2020 12:00:00 AM
Day[151] = 11/14/2020 12:00:00 AM
Day[152] = 11/15/2020 12:00:00 AM
Day[153] = 11/16/2020 12:00:00 AM
Day[154] = 11/17/2020 12:00:00 AM
Day[155] = 11/18/2020 12:00:00 AM
Day[156] = 11/19/2020 12:00:00 AM
Day[157] = 11/20/2020 12:00:00 AM
Day[158] = 11/21/2020 12:00:00 AM
Day[159] = 11/22/2020 12:00:00 AM
Day[160] = 11/23/2020 12:00:00 AM
Day[161] = 11/24/2020 12:00:00 AM
Day[162] = 11/25/2020 12:00:00 AM
Day[163] = 11/26/2020 12:00:00 AM
Day[164] = 11/27/2020 12:00:00 AM
Day[165] = 11/28/2020 12:00:00 AM
Day[166] = 11/29/2020 12:00:00 AM
Day[167] = 11/30/2020 12:00:00 AM
Day[168] = 12/01/2020 12:00:00 AM
Day[169] = 12/02/2020 12:00:00 AM
Day[170] = 12/03/2020 12:00:00 AM
Day[171] = 12/04/2020 12:00:00 AM
Day[172] = 12/05/2020 12:00:00 AM
Day[173] = 12/06/2020 12:00:00 AM
Day[174] = 12/07/2020 12:00:00 AM
Day[175] = 12/08/2020 12:00:00 AM
Day[176] = 12/09/2020 12:00:00 AM
Day[177] = 12/10/2020 12:00:00 AM
Day[178] = 12/11/2020 12:00:00 AM
Day[179] = 12/12/2020 12:00:00 AM
Day[180] = 12/13/2020 12:00:00 AM
Day[181] = 12/14/2020 12:00:00 AM
Day[182] = 12/15/2020 12:00:00 AM
Day[183] = 12/16/2020 12:00:00 AM
Day[184] = 12/17/2020 12:00:00 AM
Day[185] = 12/18/2020 12:00:00 AM
Day[186] = 12/19/2020 12:00:00 AM
Day[187] = 12/20/2020 12:00:00 AM
Day[188] = 12/21/2020 12:00:00 AM
Day[189] = 12/22/2020 12:00:00 AM
Day[190] = 12/23/2020 12:00:00 AM
Day[191] = 12/24/2020 12:00:00 AM
Day[192] = 12/25/2020 12:00:00 AM
Day[193] = 12/26/2020 12:00:00 AM
Day[194] = 12/27/2020 12:00:00 AM
Day[195] = 12/28/2020 12:00:00 AM
Day[196] = 12/29/2020 12:00:00 AM
Day[197] = 12/30/2020 12:00:00 AM
Day[198] = 12/31/2020 12:00:00 AM
Day[199] = 01/01/2021 12:00:00 AM
Day[200] = 01/02/2021 12:00:00 AM
Day[201] = 01/03/2021 12:00:00 AM
Day[202] = 01/04/2021 12:00:00 AM
Day[203] = 01/05/2021 12:00:00 AM
Day[204] = 01/06/2021 12:00:00 AM
Day[205] = 01/07/2021 12:00:00 AM
Day[206] = 01/08/2021 12:00:00 AM
Day[207] = 01/09/2021 12:00:00 AM
Day[208] = 01/10/2021 12:00:00 AM
Day[209] = 01/11/2021 12:00:00 AM
Day[210] = 01/12/2021 12:00:00 AM
Day[211] = 01/13/2021 12:00:00 AM
Day[212] = 01/14/2021 12:00:00 AM
Day[213] = 01/15/2021 12:00:00 AM
Day[214] = 01/16/2021 12:00:00 AM
Day[215] = 01/17/2021 12:00:00 AM
Day[216] = 01/18/2021 12:00:00 AM
Day[217] = 01/19/2021 12:00:00 AM
Day[218] = 01/20/2021 12:00:00 AM
Day[219] = 01/21/2021 12:00:00 AM
Day[220] = 01/22/2021 12:00:00 AM
Day[221] = 01/23/2021 12:00:00 AM
Day[222] = 01/24/2021 12:00:00 AM
Day[223] = 01/25/2021 12:00:00 AM
Day[224] = 01/26/2021 12:00:00 AM
Day[225] = 01/27/2021 12:00:00 AM
Day[226] = 01/28/2021 12:00:00 AM
Day[227] = 01/29/2021 12:00:00 AM
Day[228] = 01/30/2021 12:00:00 AM
Day[229] = 01/31/2021 12:00:00 AM
Day[230] = 02/01/2021 12:00:00 AM
Day[231] = 02/02/2021 12:00:00 AM
Day[232] = 02/03/2021 12:00:00 AM
Day[233] = 02/04/2021 12:00:00 AM
Day[234] = 02/05/2021 12:00:00 AM
Day[235] = 02/06/2021 12:00:00 AM
Day[236] = 02/07/2021 12:00:00 AM
Day[237] = 02/08/2021 12:00:00 AM
Day[238] = 02/09/2021 12:00:00 AM
Day[239] = 02/10/2021 12:00:00 AM
Day[240] = 02/11/2021 12:00:00 AM
Day[241] = 02/12/2021 12:00:00 AM
Day[242] = 02/13/2021 12:00:00 AM
Day[243] = 02/14/2021 12:00:00 AM
Day[244] = 02/15/2021 12:00:00 AM
Day[245] = 02/16/2021 12:00:00 AM
Day[246] = 02/17/2021 12:00:00 AM
Day[247] = 02/18/2021 12:00:00 AM
Day[248] = 02/19/2021 12:00:00 AM
Day[249] = 02/20/2021 12:00:00 AM
Day[250] = 02/21/2021 12:00:00 AM
Day[251] = 02/22/2021 12:00:00 AM
Day[252] = 02/23/2021 12:00:00 AM
Day[253] = 02/24/2021 12:00:00 AM
Day[254] = 02/25/2021 12:00:00 AM
Day[255] = 02/26/2021 12:00:00 AM
Day[256] = 02/27/2021 12:00:00 AM
Day[257] = 02/28/2021 12:00:00 AM
Day[258] = 03/01/2021 12:00:00 AM
Day[259] = 03/02/2021 12:00:00 AM
Day[260] = 03/03/2021 12:00:00 AM
Day[261] = 03/04/2021 12:00:00 AM
Day[262] = 03/05/2021 12:00:00 AM
Day[263] = 03/06/2021 12:00:00 AM
Day[264] = 03/07/2021 12:00:00 AM
Day[265] = 03/08/2021 12:00:00 AM
Day[266] = 03/09/2021 12:00:00 AM
Day[267] = 03/10/2021 12:00:00 AM
Day[268] = 03/11/2021 12:00:00 AM
Day[269] = 03/12/2021 12:00:00 AM
Day[270] = 03/13/2021 12:00:00 AM
Day[271] = 03/14/2021 12:00:00 AM
Day[272] = 03/15/2021 12:00:00 AM
Day[273] = 03/16/2021 12:00:00 AM
Day[274] = 03/17/2021 12:00:00 AM
Day[275] = 03/18/2021 12:00:00 AM
Day[276] = 03/19/2021 12:00:00 AM
Day[277] = 03/20/2021 12:00:00 AM
Day[278] = 03/21/2021 12:00:00 AM
Day[279] = 03/22/2021 12:00:00 AM
Day[280] = 03/23/2021 12:00:00 AM
Day[281] = 03/24/2021 12:00:00 AM
Day[282] = 03/25/2021 12:00:00 AM
Day[283] = 03/26/2021 12:00:00 AM
Day[284] = 03/27/2021 12:00:00 AM
Day[285] = 03/28/2021 12:00:00 AM
Day[286] = 03/29/2021 12:00:00 AM
Day[287] = 03/30/2021 12:00:00 AM
Day[288] = 03/31/2021 12:00:00 AM
Day[289] = 04/01/2021 12:00:00 AM
Day[290] = 04/02/2021 12:00:00 AM
Day[291] = 04/03/2021 12:00:00 AM
Day[292] = 04/04/2021 12:00:00 AM
Day[293] = 04/05/2021 12:00:00 AM
Day[294] = 04/06/2021 12:00:00 AM
Day[295] = 04/07/2021 12:00:00 AM
Day[296] = 04/08/2021 12:00:00 AM
Day[297] = 04/09/2021 12:00:00 AM
Day[298] = 04/10/2021 12:00:00 AM
Day[299] = 04/11/2021 12:00:00 AM
Day[300] = 04/12/2021 12:00:00 AM
Day[301] = 04/13/2021 12:00:00 AM
Day[302] = 04/14/2021 12:00:00 AM
Day[303] = 04/15/2021 12:00:00 AM
Day[304] = 04/16/2021 12:00:00 AM
Day[305] = 04/17/2021 12:00:00 AM
Day[306] = 04/18/2021 12:00:00 AM
Day[307] = 04/19/2021 12:00:00 AM
Day[308] = 04/20/2021 12:00:00 AM
Day[309] = 04/21/2021 12:00:00 AM
Day[310] = 04/22/2021 12:00:00 AM
Day[311] = 04/23/2021 12:00:00 AM
Day[312] = 04/24/2021 12:00:00 AM
Day[313] = 04/25/2021 12:00:00 AM
Day[314] = 04/26/2021 12:00:00 AM
Day[315] = 04/27/2021 12:00:00 AM
Day[316] = 04/28/2021 12:00:00 AM
Day[317] = 04/29/2021 12:00:00 AM
Day[318] = 04/30/2021 12:00:00 AM
Day[319] = 05/01/2021 12:00:00 AM
Day[320] = 05/02/2021 12:00:00 AM
Day[321] = 05/03/2021 12:00:00 AM
Day[322] = 05/04/2021 12:00:00 AM
Day[323] = 05/05/2021 12:00:00 AM
Day[324] = 05/06/2021 12:00:00 AM
Day[325] = 05/07/2021 12:00:00 AM
Day[326] = 05/08/2021 12:00:00 AM
Day[327] = 05/09/2021 12:00:00 AM
Day[328] = 05/10/2021 12:00:00 AM
Day[329] = 05/11/2021 12:00:00 AM
Day[330] = 05/12/2021 12:00:00 AM
Day[331] = 05/13/2021 12:00:00 AM
Day[332] = 05/14/2021 12:00:00 AM
Day[333] = 05/15/2021 12:00:00 AM
Day[334] = 05/16/2021 12:00:00 AM
Day[335] = 05/17/2021 12:00:00 AM
Day[336] = 05/18/2021 12:00:00 AM
Day[337] = 05/19/2021 12:00:00 AM
Day[338] = 05/20/2021 12:00:00 AM
Day[339] = 05/21/2021 12:00:00 AM
Day[340] = 05/22/2021 12:00:00 AM
Day[341] = 05/23/2021 12:00:00 AM
Day[342] = 05/24/2021 12:00:00 AM
Day[343] = 05/25/2021 12:00:00 AM
Day[344] = 05/26/2021 12:00:00 AM
Day[345] = 05/27/2021 12:00:00 AM
Day[346] = 05/28/2021 12:00:00 AM
Day[347] = 05/29/2021 12:00:00 AM
Day[348] = 05/30/2021 12:00:00 AM
Day[349] = 05/31/2021 12:00:00 AM
Day[350] = 06/01/2021 12:00:00 AM
Day[351] = 06/02/2021 12:00:00 AM
Day[352] = 06/03/2021 12:00:00 AM
Day[353] = 06/04/2021 12:00:00 AM
Day[354] = 06/05/2021 12:00:00 AM
Day[355] = 06/06/2021 12:00:00 AM
Day[356] = 06/07/2021 12:00:00 AM
Day[357] = 06/08/2021 12:00:00 AM
Day[358] = 06/09/2021 12:00:00 AM
Day[359] = 06/10/2021 12:00:00 AM
Day[360] = 06/11/2021 12:00:00 AM
Day[361] = 06/12/2021 12:00:00 AM
Day[362] = 06/13/2021 12:00:00 AM
Day[363] = 06/14/2021 12:00:00 AM
Day[364] = 06/15/2021 12:00:00 AM
